libs/model/__tests__/seed/seed.spec.ts

```diff
@@ -5,106 +5,42 @@ import {
+  schemas,
+  type DeepPartial,
 } from '@hicommonwealth/core';
 import { expect } from 'chai';
 import { step } from 'mocha-steps';
+import { Model, ModelStatic } from 'sequelize';
 import z from 'zod';
-import {
-  AddressSchema,
-  ChainNodeSchema,
-  CommunityContractSchema,
-  CommunitySchema,
-  CommunityStakeSchema,
-  ContractSchema,
-  GroupSchema,
-  NotificationCategorySchema,
-  ProfileSchema,
-  SchemaWithModel,
-  SeedOptions,
-  SnapshotProposalSchema,
-  SnapshotSpaceSchema,
-  SubscriptionSchema,
-  TopicSchema,
-  UserSchema,
-  checkDb,
-  seed,
-} from '../../src/test';
+import { models } from '../../src/database';
+import { SeedOptions, checkDb, seed } from '../../src/test';
 
 // testSeed creates an entity using the `seed` function
 // then attempts to find the entity and validate it
-async function testSeed<T extends SchemaWithModel<any>>(
-  schemaModel: T, // zod schema
-  overrides: Partial<z.infer<T['schema']>> = {}, // override attributes
-  options?: {
-    seedOptions?: SeedOptions;
-    testOptions?: {
-      // options for test
-      excludeOverrideChecks?: (keyof typeof overrides)[];
-    };
-  },
-) {
-  // create entity
-  const createdEntity = await seed(
-    schemaModel,
-    overrides,
-    options?.seedOptions,
-  );
-  expect(createdEntity, 'failed to create entity').not.to.be.null;
+async function testSeed<T extends schemas.Aggregates>(
+  name: T,
+  values?: DeepPartial<z.infer<typeof schemas.entities[T]>>,
+  options: SeedOptions = { mock: true },
+): Promise<z.infer<typeof schemas.entities[T]>> {
+  const [record, records] = await seed(name, values, options);
+  expect(records.length, 'failed to create entity').to.be.gt(0);
 
   // perform schema validation on created entity (throws)
-  const data = await schemaModel.schema.parse(createdEntity.toJSON());
-
-  // build query that will find the entity, or default to using `id`
-  const findQuery: Partial<z.infer<T['schema']>> = (() => {
-    if (schemaModel.buildQuery) {
-      // not all entities have a single primary key,
-      // so build a query for it
-      const q = schemaModel.buildQuery(data);
-      const attributes = Object.keys(q.where);
-      for (const attribute of attributes) {
-        expect(data, 'created entity is missing id').to.have.property(
-          attribute,
-        );
-      }
-      return q;
-    } else {
-      // use `id` as default primary key
-      expect(data, 'created entity is missing id').to.have.property('id');
-      return {
-        where: {
-          id: data.id,
-        },
-      };
-    }
-  })();
+  const schema = schemas.entities[name];
+  const model = models[name];
+  const data = await schema.parse(record);
 
   // attempt to find entity that was created
-  const existingEntity = await schemaModel.model.findOne(findQuery);
+  const existingEntity = await (model as ModelStatic<Model>).findOne({
+    where: {
+      [model.primaryKeyAttribute]: data[model.primaryKeyAttribute],
+    },
+  });
```

Comment:
My main concern here is for entities that have a composite primary key like CommunityStake.

A possible solution is to add an ID column for CommunityStakes. Oddly, the schema has an `id` field, yet the table doesn't.

Reply:
We can deal with those cases by updating the way seed traverses the graph, either by handling the special cases or by supporting composite keys in the navigation .... all that metadata is available in the models.